<a href="https://colab.research.google.com/github/EmanAfzal16/Eman-flyrank-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/EmanAfzal16/Eman-flyrank-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane: Refresh / Content Opportunity Scoring

I selected the **Refresh / Content Opportunity Scoring** lane because it addresses a practical SEO challenge: identifying which pages deserve attention first. Large websites often contain thousands of pages, making manual review inefficient and inconsistent. By analyzing observable search and content signals, this project aims to prioritize pages with the highest potential for improvement. I chose this lane because it combines data-driven decision making with real business value, helping teams focus their efforts where they are most likely to make a meaningful impact.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

**Research Question:** Which pages should be prioritized for content refresh based on observable search and content signals?

**Decision:** Determine which pages deserve immediate review instead of treating all pages equally.

**Who Acts:** SEO specialists, content strategists, or website managers responsible for maintaining search performance.

**Action:** Review the recommended pages, update outdated content, improve relevance, or monitor pages that do not require immediate changes.

**Cost of a Wrong Recommendation:** If a high-potential page is overlooked, valuable optimization opportunities may be missed. If low-impact pages are prioritized, time and resources are wasted, reducing the overall efficiency of the content optimization process.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [25]:
!git clone https://github.com/EmanAfzal16/Eman-flyrank-internship.git
%cd Eman-flyrank-internship

Cloning into 'Eman-flyrank-internship'...
remote: Enumerating objects: 119, done.
remote: Counting objects: 100% (119/119), done.
remote: Compressing objects: 100% (76/76), done.
remote: Total 119 (delta 35), reused 94 (delta 27), pack-reused 0 (from 0)
Receiving objects: 100% (119/119), 1.84 MiB | 10.37 MiB/s, done.
Resolving deltas: 100% (35/35), done.
/content/Eman-flyrank-internship/Eman-flyrank-internship/Eman-flyrank-internship/Eman-flyrank-internship


In [26]:
!ls data/raw

content_refresh_anonymized.csv


In [27]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total Pages:", len(df))
print("Total Columns:", len(df.columns))
print("Average CTR:", round(df["ctr"].mean(), 3))
print("Average Impressions:", round(df["impressions_90d"].mean(), 2))

Total Pages: 30000
Total Columns: 44
Average CTR: 0.511
Average Impressions: 5200.37


## Quick Look at the Data

The starter dataset contains **30,000 pages** and **44 features**, providing enough information to analyze search performance and content quality. The average CTR is **0.511**, indicating that user engagement varies across pages, while the average number of impressions is **5,200.37**, showing that many pages receive substantial search visibility. These observations suggest that prioritizing which pages need content updates or optimization is a meaningful problem, making this research lane worth exploring over the next seven weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words: What I Can and Can't Claim

This project aims to identify meaningful patterns in search and content data that can support better content optimization decisions. Any findings will be based on observed data and should be interpreted as directional insights rather than definitive facts. The results may help prioritize which pages deserve attention, but they cannot guarantee improved rankings or prove that one factor directly causes another. This work is designed to assist decision-making using evidence from the dataset, not to predict Google's ranking algorithm or replace human judgment.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.